packages/react/src/lib/useVoiceClient.ts

```diff
@@ -92,14 +105,41 @@ export const useVoiceClient = (props: {
           onMessage.current?.(message);
         }
 
-        if (message.type === 'tool_call') {
-          const response = await onToolCall.current?.(message);
-          if (response) {
-            const parsed = ToolResponseSchema.safeParse(response);
-            if (parsed.success) {
-              client.current?.sendToolResponse(parsed.data);
-            }
-          }
+        if (message.type === 'tool_call' && onToolCall.current) {
+          void onToolCall
+            .current(message, {
+              success: (content: unknown) => ({
+                type: 'tool_response',
+                tool_call_id: message.tool_call_id,
+                content: JSON.stringify(content),
+              }),
+              error: ({
+                error,
+                code,
+                level,
+                content,
+              }: {
+                error: string;
+                code: string;
+                level: string;
+                content: string;
+              }) => ({
+                type: 'tool_error',
+                tool_call_id: message.tool_call_id,
+                error,
+                code,
+                level,
+                content,
+              }),
+            })
+            .then((response) => {
+              const parsed = ToolResponseSchema.safeParse(response);
+              if (parsed.success) {
+                client.current?.sendToolResponse(parsed.data);
```

Comment:
should make this `sendToolError` if it's an error or remove the `sendToolError` method

Reply:
@hume-brian oh, this a block you added in #188.

https://github.com/HumeAI/empathic-voice-api-js/pull/188/files#diff-606d0dd18ceb7429e2c5d6095b5a67db2448400e9ba863976beeaac5cfafc32cR98

@yinishi do you think this should be something like 

```typescript
z.union([ToolResponseSchema, ToolErrorSchema]).safeParse(response)
```

Reply:
@hume-brian @yinishi I updated the zod parsing and function name :)

Reply:
lgtm!